### ISSUE TITLE
How to pass arg to Puppeteer via decktape command line?

### ISSUE THREAD

**Issue by gitpitch** [astefanutti/decktape#130]
Not sure if this is a bug or I'm simply doing it incorrectly but I am trying to pass the `--allow-file-access-from-files` arg on the decktape command line. My hope is to see the arg passed on to Puppeteer.launch() which would then configure the underlying Chrome instance for the behavior that I need.

I tried the following on the command line:

```
/bin/decktape reveal /home/index.html ./out.pdf --allow-file-access-from-files
```

That resulted in the following error message in the console output:

```
'--allow-file-access-from-files' expects a value
```

You can see it fails with the `expects a value` error message. I tracked that message down to `lib/nonnom.js` which is parsing/processing the args list.

I also tried to place the arg at the start as follows:

```
$ /bin/decktape --allow-file-access-from-files reveal /home/index.html ./out.pdf

command 'reveal' is expected to be the first argument
```
As you can see, this too results in an error.

Given I could not get this to work I decided to make the following tweak to `decktape.js` to see if I could bypass args passing/processing and just manually configure Puppeteer for the args that I want:

```
(async () => {

  const browser = await puppeteer.launch({
    headless : true,
    executablePath: options.executablePath,
    /*
    args     : Object.keys(options).reduce((args, option) => {
      switch (option) {
        case 'sandbox':
          if (options.sandbox === false) args.push('--no-sandbox');
          break;
      }
      return args;
    }, [])
    */
    args: [ '--allow-file-access-from-files' ],
  });
  const page = await browser.newPage();
  await page.emulateMedia('screen');
  const printer = hummus.createWriter(options.filename);
  const metadata = printer.getDocumentContext().getInfoDictionary();
  metadata.creator = 'Decktape';
```

You can see above that I commented out the original args parsing code block and replaced it with a static args array that contains the command line arg I want to pass to Puppeteer:

```
    args: [ '--allow-file-access-from-files' ]
```

This change to `decktape.js` gave me exactly the behavior that I want from Puppeteer/Chrome when I executed the following on the command line:

```
/bin/decktape reveal /home/index.html ./out.pdf
```

So my question is this, what is the correct syntax using the official `decktape.js` in order to execute 
my command that would ensure the `--allow-file-access-from-files` arg would be passed through to Puppeteer?

Thanks for any help you can provide.

**Comment by astefanutti** [comment]
For the time being, DeckTape only passes these two white-listed options:
https://github.com/astefanutti/decktape/blob/5669c5d78917950850df6d34a8f539bd50470ef0/decktape.js#L87-L98
It'd be valuable to extend it to the complete list of Puppeteer options. I'd be incline to keep it as a white list, and find a way to retrieve that list programmatically ideally.

**Comment by gitpitch** [comment]
Thanks for this feedback. I guess the complete list of Puppeteer options would be at least as long as the complete list of Chrome/Chromium options. I found [this list](https://peter.sh/experiments/chromium-command-line-switches/) which documents Chromium options. As you can see it's very, very extensive.

I'm curious what are you protecting users from by whitelisting options on the decktape command line? Basically, why did you select these two options in particular from a huge list of options?

And I guess I should ask, If I created a PR that added `--allow-file-access-from-files` to your existing whitelist would you accept it?

**Comment by astefanutti** [comment]
> I'm curious what are you protecting users from by whitelisting options on the decktape command line? Basically, why did you select these two options in particular from a huge list of options?

I needed that two options for the Docker image 😉. One alternative would be to pass all the unknown options as Puppeteer args... Or to introduce a option that could be an array or multi-evaluated, e.g.:

    $ decktape URL FILE --puppeteer-arg=--allow-file-access-from-files --puppeteer-arg=--no-sandbox]
    $ decktape URL FILE --puppeteer-args='[--allow-file-access-from-files, --no-sandbox]'

> And I guess I should ask, If I created a PR that added --allow-file-access-from-files to your existing whitelist would you accept it?

Sure!

**Comment by astefanutti** [comment]
Fixed with 6b46a957801ddc518133df24a9c73acf6bc968f5.
